publisher/options.go

```diff
@@ -0,0 +1,57 @@
+package publisher
+
+import (
+	"strconv"
+	"time"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+// Option allows to configure RabbitMQ Publisher.
+type Option func(p *Publisher)
+
+// WithConstHeaders sets constant publishing headers, that will be added to all publishings.
+func WithConstHeaders(headers amqp.Table) Option {
+	return func(p *Publisher) {
+		if p.headers == nil {
+			p.headers = make(amqp.Table)
+		}
+
+		for k, v := range headers {
+			p.headers[k] = v
+		}
+	}
+}
+
+// WithTransientDeliveryMode sets publishing to the Transient delivery mode.
+// Transient means higher throughput but messages will not be
+// restored on broker restart.
+// See https://github.com/rabbitmq/amqp091-go/blob/main/types.go#L123.
+func WithTransientDeliveryMode() Option {
+	return func(p *Publisher) {
+		p.deliveryMode = amqp.Transient
+	}
+}
+
+// WithMandatory sets server to discard a message if no queue is
```

Comment:
from the [documentation](https://www.rabbitmq.com/amqp-0-9-1-reference.html#basic.publish.immediate):

> This flag tells the server how to react if the message cannot be routed to a queue consumer immediately. If this flag is set, the server will return an undeliverable message with a Return method. If this flag is zero, the server will queue the message, but with no guarantee that it will ever be consumed.

The message is discarded but what is more important the publisher is notified about this situation.

Reply:
thanks for the link, it is updated

Reply:
What about adding an example of how to use this flag and listen for undeliverable messages?

Reply:
I think it out of the scope of this package. The scope of this package is to help with reconnections and simplify creation of consumers/publishers, not how to use channel

Reply:
ok